front-end-tools/src/components/WriteComponent.tsx

```diff
@@ -0,0 +1,618 @@
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import Select from 'react-select';
+import { Alert, Button, Form, Row } from 'react-bootstrap';
+
+import { WalletConnection } from '@concordium/react-components';
+import {
+    ModuleReference,
+    TransactionKindString,
+    TransactionSummaryType,
+    displayTypeSchemaTemplate,
+    toBuffer,
+    getUpdateContractParameterSchema,
+    ConcordiumGRPCClient,
+} from '@concordium/web-sdk';
+
+import Box from './Box';
+import { TxHashLink } from './CCDScanLinks';
+import { write } from '../writing_to_blockchain';
+import { getEmbeddedSchema, getContractInfo } from '../reading_from_blockchain';
+import { getObjectExample, getArrayExample } from '../utils';
+import { REFRESH_INTERVAL, INPUT_PARAMETER_TYPES_OPTIONS } from '../constants';
+
+interface ConnectionProps {
+    isTestnet: boolean;
+    account: string;
+    connection: WalletConnection;
+    client: ConcordiumGRPCClient | undefined;
+}
+
+export default function WriteComponenet(props: ConnectionProps) {
+    const { isTestnet, account, connection, client } = props;
+
+    type FormType = {
+        smartContractIndex: number;
+        smartContractName: string | undefined;
+        entryPointName: string | undefined;
+        file: FileList | undefined;
+        hasInputParameter: boolean;
+        deriveFromSmartContractIndex: boolean;
+        inputParameterType: string | undefined;
+        inputParameter: string | undefined;
+        maxExecutionEnergy: number;
+        isPayable: boolean;
+        cCDAmount: number;
+    };
+
+    const form = useForm<FormType>();
+    const deriveContractInfo = form.watch('deriveFromSmartContractIndex');
+    const hasInputParameter = form.watch('hasInputParameter');
+    const entryPointName = form.watch('entryPointName');
+    const smartContractName = form.watch('smartContractName');
+    const inputParameterType = form.watch('inputParameterType');
+
+    const isPayable = form.watch('isPayable');
```

Comment:
Would be nice to refactor this to [`useWatch`](https://react-hook-form.com/docs/usewatch) as well, as it is recommended to use to trigger side effects.